scripts/lint.js

```diff
@@ -1,53 +1,90 @@
 const chalk = require('chalk');
 const baseESlintConfig = require('eslint-config-seek');
 const EslintCLI = require('eslint').CLIEngine;
-const builds = require('../config/builds');
+const lodash = require('lodash');
+const glob = require('glob-promise');
 
+const builds = require('../config/builds');
 const prettierCheck = require('../lib/runPrettier').check;
 const prettierConfig = require('../config/prettier/prettierConfig');
-
-const eslintDefaultPath = require('../config/eslint/defaultPath');
+const runTsc = require('../lib/runTsc');
 
 const args = require('../config/args').argv;
 
-console.log(chalk.cyan('Linting'));
+const run = async () => {
+  console.log(chalk.cyan('Linting'));
 
-// Decorate eslint config is not supported for monorepo
-const eslintConfig =
-  builds.length === 1
-    ? builds[0].eslintDecorator(baseESlintConfig)
-    : baseESlintConfig;
+  const hasTSFiles = await glob('**/*.{ts,tsx}').then(
+    files => files.length > 0
+  );
 
-const cli = new EslintCLI({
-  baseConfig: eslintConfig,
-  useEslintrc: false
-});
+  if (hasTSFiles) {
+    try {
+      await runTsc();
```

Comment:
Not sure about `tsc` here. The TypeScript projects I've worked on follow a pattern I've found common with other compiled languages where you compile first, which catches compile errors, then lint. Running `tsc` as a lint tool would mean that a prod build does the Typescript compilation twice.

That said, the boundary between `tsc` and `(t|e)slint` is confusing; for example it's `tsc` that catches things like null safety and the other `strict` settings, but I tend to think of those as linting checks.

Reply:
We're taking a different approach here. Babel is still our compilation tool and we are using (@babel/preset-typescript)[https://babeljs.io/docs/en/babel-preset-typescript] to handle the typescript part. We run `tsc` as part of the lint phase to ensure our types are correct. We will probably add `tslint` to this as well but didn't want to overwhelm this PR.

Reply:
To be clear, `tsc` has no part in our compilation/bundling. It's only used to validate types. This is great though as we can have the best of both worlds.

Reply:
Didn't realise babel doesn't do type checking, so in that context it makes a lot of sense to run `tsc` here 👍